From a052fe16ee430847e2453dbb9684bce47da2402a Mon Sep 17 00:00:00 2001
From: Will Dietz <w@wdtz.org>
Date: Thu, 6 Oct 2016 14:09:59 -0500
Subject: [PATCH 7/8] Partially revert "remove potentially PIC-incompatible
 relocations from x86_64 and x32 asm"

This partially reverts commit b35c4c475bea3c8f938d8e9696d1138eabb54a89.
---
 src/math/x32/expl.s           | 2 +-
 src/math/x86_64/expl.s        | 2 +-
 src/signal/x32/sigsetjmp.s    | 4 ++--
 src/signal/x86_64/sigsetjmp.s | 4 ++--
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/math/x32/expl.s b/src/math/x32/expl.s
index 369f7bd2..aabea45e 100644
--- a/src/math/x32/expl.s
+++ b/src/math/x32/expl.s
@@ -37,7 +37,7 @@ expl:
 	fstpt (%esp)
 	fstpt 16(%esp)
 	fstpt 32(%esp)
-	call exp2l@PLT
+	call exp2l
 		# if 2^hi == inf return 2^hi
 	fld %st(0)
 	fstpt (%esp)
diff --git a/src/math/x86_64/expl.s b/src/math/x86_64/expl.s
index 798261d2..3add810d 100644
--- a/src/math/x86_64/expl.s
+++ b/src/math/x86_64/expl.s
@@ -37,7 +37,7 @@ expl:
 	fstpt (%rsp)
 	fstpt 16(%rsp)
 	fstpt 32(%rsp)
-	call exp2l@PLT
+	call exp2l
 		# if 2^hi == inf return 2^hi
 	fld %st(0)
 	fstpt (%rsp)
diff --git a/src/signal/x32/sigsetjmp.s b/src/signal/x32/sigsetjmp.s
index 1f02b0e5..bb1c245e 100644
--- a/src/signal/x32/sigsetjmp.s
+++ b/src/signal/x32/sigsetjmp.s
@@ -11,7 +11,7 @@ __sigsetjmp:
 	mov %rbx,72+8(%rdi)
 	mov %rdi,%rbx
 
-	call setjmp@PLT
+	call setjmp
 
 	pushq 64(%rbx)
 	movl $0, 4(%rsp)
@@ -22,4 +22,4 @@ __sigsetjmp:
 .hidden __sigsetjmp_tail
 	jmp __sigsetjmp_tail
 
-1:	jmp setjmp@PLT
+1:	jmp setjmp
diff --git a/src/signal/x86_64/sigsetjmp.s b/src/signal/x86_64/sigsetjmp.s
index 9a7695f9..d354d680 100644
--- a/src/signal/x86_64/sigsetjmp.s
+++ b/src/signal/x86_64/sigsetjmp.s
@@ -11,7 +11,7 @@ __sigsetjmp:
 	mov %rbx,72+8(%rdi)
 	mov %rdi,%rbx
 
-	call setjmp@PLT
+	call setjmp
 
 	pushq 64(%rbx)
 	mov %rbx,%rdi
@@ -21,4 +21,4 @@ __sigsetjmp:
 .hidden __sigsetjmp_tail
 	jmp __sigsetjmp_tail
 
-1:	jmp setjmp@PLT
+1:	jmp setjmp
-- 
2.11.0

